pylsp/python_lsp.py

```diff
@@ -162,8 +162,8 @@ def capabilities(self):
                 'resolveProvider': False,  # We may need to make this configurable
             },
             'completionProvider': {
-                'resolveProvider': False,  # We know everything ahead of time
-                'triggerCharacters': ['.']
+                'resolveProvider': True,  # We could know everything ahead of time, but this takes time to transfer
```

Comment:
If this value is True, then it means that `eager` is False, right? Then if a client decides to have all the information ready and sets `eager` to True, then it is their responsibility to not calling `completionResolve`?

Reply:
> If this value is True, then it means that eager is False, right?

IMO no, `eager` still can be `True`. `resolveProvider` just declares that we can handle

`resolveProvider` signals that a server will handle `completionItem/resolve` requests; it does not forbid server to return resolved items in the first place.

> if a client decides to have all the information ready and sets `eager` to True, then it is their responsibility to not calling `completionResolve`?

No, the client is still allowed to call `completionItem/resolve`, it will be just inefficient for them to do so.